src/main/scala/treadle/executable/VcdMemoryLoggingController.scala

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Regents of the University of California (Regents)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package treadle.executable
+
+import firrtl.MemKind
+import logger.LazyLogging
+import treadle.utils.NumberHelpers
+
+import scala.util.matching.Regex
+
+/** Controls whether a given memory cell should be logged to vcd output
+  * if logAllRadixOpt is defined then all indices for all memories should be logged
+  * otherwise if info has an entry then check the memory index
+  * if the set contains -1 then all indices should be logged for that memory
+  *
+  * @param logAllRadixOpt  if defined then log all memories using this radix
+  * @param memoriesTracked    map of memories to the set of indices that should be logged
+  */
+class VcdMemoryLoggingController(
+  logAllRadixOpt:  Option[Int] = None,
+  memoriesTracked: Map[Symbol, IndicesAndRadix] = Map.empty
```

Comment:
is `memoriesTracked` ignored when `logAllRadixOpt` is `Some`?
This seems to be the case in `vcdKey` but not in `getIndexedName`. Why?

Reply:
Great catch. This is only used for building the directory in the VCD header.